worlds/marioland2/__init__.py

```diff
@@ -0,0 +1,374 @@
+import base64
+
+import Utils
+import settings
+
+from worlds.AutoWorld import World, WebWorld
+from BaseClasses import Region, Location, Item, ItemClassification, Tutorial
+
+from . import client
+from .rom import generate_output, SuperMarioLand2DeltaPatch
+from .options import SML2Options
+from .locations import locations
+from .items import items
+from .logic import has_pipe_up, has_pipe_down, has_pipe_left, has_pipe_right, has_level_progression
+
+START_IDS = 7770000
+
+
+class MarioLand2Settings(settings.Group):
+    class SML2RomFile(settings.UserFilePath):
+        """File name of the Super Mario Land 2 1.0 ROM"""
+        description = "Super Mario Land 2 - 6 Golden Coins (USA, Europe) 1.0 ROM File"
+        copy_to = "Super Mario Land 2 - 6 Golden Coins (USA, Europe).gb"
+        md5s = [SuperMarioLand2DeltaPatch.hash]
+
+    rom_file: SML2RomFile = SML2RomFile(SML2RomFile.copy_to)
+
+
+class MarioLand2WebWorld(WebWorld):
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to playing Super Mario Land 2 with Archipelago.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Alchav"]
+    )
+
+    tutorials = [setup_en]
+
+
+class MarioLand2World(World):
+    """Super Mario Land 2 is a classic platformer that follows Mario on a quest to reclaim his castle from the
+    villainous Wario. This iconic game features 32 levels, unique power-ups, and introduces Wario as Mario's
+    arch-rival."""  # -ChatGPT
```

Comment:
You should probably write something yourself

Reply:
no u